### ISSUE TITLE
Automatically declare ins variables AND buffers

### ISSUE THREAD

**Issue by vitreo12** [vitreo12/omni#93]
```nim
ins 3:
    var1
    var2 {Buffer}
    var3

sample:
    out1 = var1 + var2[0] + var3
```

**Comment by vitreo12** [comment]
1. `Buffer` ins should error out if accessed in `init`
2. `Buffer` can only be accessed in `perform` / `sample`

**Comment by vitreo12** [comment]
Should there be access to these variables in `defs` too? It would be pretty neat, but it would require passing the `ins_Nim` pointers around.

**Comment by vitreo12** [comment]
Merged with #100

**Comment by vitreo12** [comment]
PR at #108
